pyproject.toml

```diff
@@ -54,3 +54,10 @@ order-by-type = false
 [tool.pyright]
 reportIncompatibleMethodOverride = true
 include = ["quax", "tests"]
+
+[dependency-groups]
+dev = [
+    "ipykernel>=6.29.5",
+    "pre-commit>=4.1.0",
+    "pytest>=8.3.4",
+]
```

Comment:
Very happy to move over to this if you'd like, although if we do then I think we should remove `tests/requirements.txt` and fold that into here as well, updating the CI and CD appropriately.

Reply:
Ok. I'll do this in a separate PR.
I'll make `test` and `docs` dep-groups and switch the CI/CD to `uv`.

Reply:
Removing from here.

Reply:
Done.